scaffoldmaker/meshtypes/meshtype_3d_solidsphere1.py

```diff
@@ -0,0 +1,510 @@
+                        -1.0,
+                        math.cos(radiansAround), math.sin(radiansAround), radiansPerElementAround,
+                        math.cos(radiansAroundNext), math.sin(radiansAroundNext), radiansPerElementAround,
+                        math.cos(radiansAround), math.sin(radiansAround), radiansPerElementAround, math.cos(radiansInclineNext), math.sin(radiansInclineNext),
+                        math.cos(radiansAroundNext), math.sin(radiansAroundNext), radiansPerElementAround, math.cos(radiansInclineNext), math.sin(radiansInclineNext)
+                    ]
+                    result2 = element.setScaleFactors(eft3, scalefactors)
+                    # print('Tetrahedron top element', elementIdentifier, result1, result2, nodeIdentifiers)
+                    elementIdentifier = elementIdentifier + 1
+
+            else:
+                # Create pyramid elements on the top pole
+                bni5 = elementsCountUp + 1 + (e3-1)*no3 + (elementsCountUp-2) * no2 + 1
+                radiansIncline = math.pi*0.5*e3/elementsCountRadial
+                radiansInclineNext = math.pi*0.5*(e3 + 1)/elementsCountRadial
+
+                for e1 in range(elementsCountAround):
+                    va = e1
+                    vb = (e1 + 1)%elementsCountAround
+                    eft5 = tricubichermite.createEftPyramidTop(va*100, vb*100, 100000 + e3*2)
+
+                    elementtemplate5.defineField(coordinates, -1, eft5)
+                    element = mesh.createElement(elementIdentifier, elementtemplate5)
+                    nodeIdentifiers = [ bni5 + va, bni5 + vb, elementsCountUp + 1, bni5 + no3 + va, bni5 + no3 + vb ]
+                    result1 = element.setNodesByIdentifier(eft5, nodeIdentifiers)
+                    # set general linear map coefficients
+                    radiansAround = va*radiansPerElementAround
+                    radiansAroundNext = vb*radiansPerElementAround
+                    scalefactors = [
+                        -1.0,
+                        math.cos(radiansAround), math.sin(radiansAround), radiansPerElementAround, math.cos(radiansIncline), math.sin(radiansIncline),
+                        math.cos(radiansAroundNext), math.sin(radiansAroundNext), radiansPerElementAround, math.cos(radiansIncline), math.sin(radiansIncline),
+                        math.cos(radiansAround), math.sin(radiansAround), radiansPerElementAround, math.cos(radiansInclineNext), math.sin(radiansInclineNext),
+                        math.cos(radiansAroundNext), math.sin(radiansAroundNext), radiansPerElementAround, math.cos(radiansInclineNext), math.sin(radiansInclineNext)
+                    ]
+                    result2 = element.setScaleFactors(eft5, scalefactors)
+                    # print('pyramid top element', elementIdentifier, result1, result2, nodeIdentifiers)
+                    elementIdentifier = elementIdentifier + 1
+
+        fm.endChange()
+
+    @staticmethod
```

Comment:
Change to:
    @classmethod
    def generateMesh(cls, region, options):
Makes it easy to call cls.generateBaseMesh inside.